routes/challenge.js

```diff
@@ -1,81 +1,104 @@
-import express from 'express';
-import bodyParser from 'body-parser';
+import express from "express";
+import bodyParser from "body-parser";
 
-import { Log } from '../models/Log.js';
-import { Challenge } from '../models/Challenge.js';
-import { User } from '../models/User.js';
-import isAuthenticated from '../config/middleware.js';
+import { Log } from "../models/Log.js";
+import { Challenge } from "../models/Challenge.js";
+import { User } from "../models/User.js";
+import isAuthenticated from "../config/middleware.js";
 
 const router = express.Router();
 
 router.use(bodyParser.json());
 router.use(bodyParser.urlencoded());
 
-router.post('/:id', isAuthenticated, async (req, res) => {
-    const id = req.params.id
-    if (id != 1 && id != 3) return res.sendStatus(403)
+router.post("/:id", isAuthenticated, async (req, res) => {
+  const id = req.params.id;
+  if (id != 1 && id != 3) return res.sendStatus(403);
 
-    const user = await User.findOne({user_id: req.body.auth})
-    
-    let repeat = false
-    
-    let maxSolved = 0
+  const user = await User.findOne({ user_id: req.body.auth });
 
-    user.challengesSolved.filter((x) => (x[0] == id)).forEach((x) => {
-        if (parseInt(x[1]) > maxSolved) {
-            maxSolved = parseInt(x[1])
-        }
-    });
-    
-    // if (repeat) return res.sendStatus(403)
-
-    user.currentRoom = id
-    user.currentChallenge = maxSolved
-
-    const resChallenge = await Challenge.findOne({ roomNumber: id, challengeNumber: maxSolved })
-    await user.save()
-    res.send({ 
-        question: resChallenge.question, 
-        roomNumber: resChallenge.roomNumber, 
-        challengeNumber: resChallenge.challengeNumber
-    })
-})
-    
-router.post('/:id/solve', isAuthenticated, async (req, res) => {
-    const id = req.params.id
-    const ans = req.body.answer.toLowerCase().trim().split(' ').join('');
-    const cd = req.body.challengeNumber
-
-    const challenge = await Challenge.findOne({ roomNumber: id, challengeNumber: cd })
-    const user = await User.findOne({user_id: req.body.auth})
-
-    const log = new Log({
-        username: user.username,
-        answer: ans,
-        roomNumber: id,
-        challengeNumber: cd,
-        time: new Date()
-    });
-
-    await log.save()
+  let repeat = false;
 
-    const correct = ans == challenge.answer.toLowerCase().trim().split(' ').join('');
+  let maxSolved = -1;
```

Comment:
does this mean the challenges now start from 0 on mongo?

Reply:
Yeah